apps/core/src/hooks/useGetKioskContents.ts

```diff
@@ -1,71 +1,102 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { SuiAddress, SuiObjectResponse } from '@mysten/sui.js';
+import { JsonRpcProvider, SuiAddress, SuiObjectResponse } from '@mysten/sui.js';
+import { fetchKiosk, getOwnedKiosks } from '@mysten/kiosk';
 import { useQuery } from '@tanstack/react-query';
 import { useRpcClient } from '../api/RpcClientContext';
-import { useGetOwnedObjects } from './useGetOwnedObjects';
+
+const getKioskId = (obj: SuiObjectResponse) =>
+	obj.data?.content &&
+	'fields' in obj.data.content &&
+	(obj.data.content.fields.for ?? obj.data.content.fields.kiosk);
 
 // OriginByte module for mainnet (we only support mainnet)
 export const ORIGINBYTE_KIOSK_MODULE =
 	'0x95a441d389b07437d00dd07e0b6f05f513d7659b13fd7c5d3923c7d9d847199b::ob_kiosk' as const;
 export const ORIGINBYTE_KIOSK_OWNER_TOKEN = `${ORIGINBYTE_KIOSK_MODULE}::OwnerToken`;
 
-const KIOSK_MODULE = '0x2::kiosk';
-const KIOSK_OWNER_CAP = `${KIOSK_MODULE}::KioskOwnerCap`;
+async function getOriginByteKioskContents(address: SuiAddress, rpc: JsonRpcProvider) {
+	const data = await rpc.getOwnedObjects({
+		owner: address,
+		filter: {
+			StructType: ORIGINBYTE_KIOSK_OWNER_TOKEN,
+		},
+	});
+	const ids = data.data.flatMap((object) => getKioskId(object) ?? []);
 
-const getKioskId = (obj: SuiObjectResponse) =>
-	obj.data?.content &&
-	'fields' in obj.data.content &&
-	(obj.data.content.fields.for ?? obj.data.content.fields.kiosk);
+	// fetch the user's kiosks
+	const ownedKiosks = await rpc.multiGetObjects({
+		ids: ids,
+		options: {
+			showContent: true,
+		},
+	});
 
-export function useGetKioskContents(address?: SuiAddress | null, disableOriginByteKiosk?: boolean) {
-	const rpc = useRpcClient();
-	const { data } = useGetOwnedObjects(address, {
-		MatchAny: [
-			{ StructType: KIOSK_OWNER_CAP },
-			...(!disableOriginByteKiosk ? [{ StructType: ORIGINBYTE_KIOSK_OWNER_TOKEN }] : []),
-		],
+	// find object IDs within a kiosk
+	const kioskObjectIds = await Promise.all(
+		ownedKiosks.map(async (kiosk) => {
+			if (!kiosk.data?.objectId) return [];
+			const objects = await rpc.getDynamicFields({
+				parentId: kiosk.data.objectId,
+			});
+			return objects.data.map((obj) => obj.objectId);
+		}),
+	);
+
+	// fetch the contents of the objects within a kiosk
+	const kioskContent = await rpc.multiGetObjects({
```

Comment:
Does that support multiple pages? 
Considering that we're merging all owned kiosks, sensing it's pretty easy to reach the 50 limit.

Reply:
Not at the moment, I left this as-is for now. Depdending on timing I may address in this PR or in a follow-up